src/client/app/justifications/justifications-modal.component.ts

```diff
@@ -0,0 +1,71 @@
+'use strict';
+import { Component, Input, EventEmitter, Output, ViewChild } from '@angular/core';
+import { ModalDirective } from 'ng2-bootstrap';
+
+@Component({
+	selector: 'justificationsModal',
+	templateUrl: 'justifications-modal.component.html'
+})
+export class JustificationsModalComponent {
+
+	@Input() title: string;
+
+	@Input() placeholder: string = 'Enter justification...';
+
+	@Input() okText: string = 'Submit';
+
+	@Input() cancelText: string = 'Cancel';
+
+	@Input()
+	get showModal() {
+		return this.showModalValue;
+	}
+	set showModal(v: boolean) {
+		this.showModalValue = v;
+		this.showModalChange.emit(this.showModalValue);
+
+		if (this.showModalValue) {
+			this.modal.show();
+		}
+	}
+	@Output() showModalChange = new EventEmitter<boolean>();
+
+	@Output() userSubmit = new EventEmitter<boolean>();
+
+	@ViewChild('fhJustificationsModal') private modal: ModalDirective;
```

Comment:
That's a good point, I've since changed this to `justificationsModalChild` instead.